lms/djangoapps/shoppingcart/views.py

```diff
@@ -430,11 +427,20 @@ def _is_enrollment_code_an_update(course, user, redemption_code):
 def use_registration_code(course_reg, user):
     """
     This method utilize course registration code.
+    If the registration code is invalid, it returns an error.
     If the registration code is already redeemed, it returns an error.
     Else, it identifies and removes the applicable OrderItem from the Order
     and redirects the user to the Registration code redemption page.
     """
-    if RegistrationCodeRedemption.is_registration_code_redeemed(course_reg):
+    if not course_reg.is_valid:
+        log.warning(u"The registration code '%s' is no longer valid.", course_reg.code)
+        return HttpResponseBadRequest(
+            _("The registration code '{registration_code}' is not longer valid.").format(
+                registration_code=course_reg.code
+            )
+        )
+
+    if RegistrationCodeRedemption.is_registration_code_redeemed(course_reg.code):
         log.warning(u"Registration code '%s' already used", course_reg.code)
```

Comment:
This enrollment code ({%s}) has already been used.

(Change the single quotation marks to parentheses.)